optional/abstract_optional.py

```diff
@@ -43,3 +43,6 @@ def map(self, func):
     @abstractmethod
     def flat_map(self, func):
         pass
+
+    __bool__ = is_present
+    __nonzero__ = is_present
```

Comment:
I wonder if:
```
__bool__ = is_present
__nonzero__ = __bool__
```
would make more sense here. I dislike both names (`bool` and `nonzero`) but I really dislike `nonzero`.  I wonder if this is a place where a comment might actually be valuable:
```
__nonzero__ = __bool__ # Python2 truthiness support.
```

Reply:
Or `__bool__ = __nonzero__ = is_present`, even?

Reply:
If that is possible it would be nice.